spec/commands/feed/find_new_stories_spec.rb

```diff
@@ -1,62 +1,60 @@
-      story1 = double(published: nil, id: "story1")
-      story2 = double(published: nil, id: "story2")
-      feed   = double(entries: [story1, story2])
-      time = Time.zone.local(2013, 1, 2)
-      result = described_class.call(feed, 1, time)
+      feed = create(:feed)
+      create(:story, entry_id: "www.story1.com", feed_id: feed.id)
+      raw_feed = create_raw_feed(feed, entries: [create_entry])
+      result = described_class.call(raw_feed, feed.id)
+
       expect(result).to be_empty
     end
   end
 
   context "the feed contains new stories" do
     it "returns stories that are not found in the database" do
-      story1 = double(published: nil, id: "story1")
-      story2 = double(published: nil, id: "story2")
-      feed   = double(entries: [story1, story2])
-
-      allow(StoryRepository)
-        .to receive(:exists?).with("story1", 1).and_return(true)
-      allow(StoryRepository)
-        .to receive(:exists?).with("story2", 1).and_return(false)
-
-      time = Time.zone.local(2013, 1, 2)
-      result = described_class.call(feed, 1, time)
-      expect(result).to eq([story2])
+      feed = create(:feed)
+      create(:story, entry_id: "www.story1.com", feed_id: feed.id)
+      entry = create_entry(published: Time.zone.now, url: "www.story2.com")
+      raw_feed = create_raw_feed(feed, entries: [entry])
+
+      expect(described_class.call(raw_feed, feed.id)).to eql(raw_feed.entries)
     end
   end
 
   it "scans until matching the last story id" do
-    new_story = double(published: nil, id: "new-story")
-    old_story = double(published: nil, id: "old-story")
-    feed = double(last_modified: nil, entries: [new_story, old_story])
+    new_entry = create_entry(published: Time.zone.now, url: "www.new.com")
+    old_entry = create_entry(title: "old entry", url: "www.old.com")
+    raw_feed = create_raw_feed(create(:feed), entries: [new_entry, old_entry])
+    result = described_class.call(raw_feed, 1, "www.old.com")
 
-    result = described_class.call(feed, 1, "old-story")
-    expect(result).to eq([new_story])
+    expect(result).to eq([raw_feed.entries.first])
   end
 
   it "ignores stories older than 3 days" do
-    new_stories = [
-      double(published: 1.hour.ago, id: "new-story"),
-      double(published: 2.days.ago, id: "new-story")
-    ]
-
-    stories_older_than_3_days = [
-      double(published: 3.days.ago, id: "new-story"),
-      double(published: 4.days.ago, id: "new-story")
-    ]
-
-    feed = double(
-      last_modified: nil,
-      entries: new_stories + stories_older_than_3_days
-    )
-
-    result = described_class.call(feed, 1)
-    expect(result).not_to include(stories_older_than_3_days)
+    feed = create(:feed)
+    new_entry = create_entry(published: 2.days.ago)
+    old_entry = create_entry(published: 4.days.ago)
+    raw_feed = create_raw_feed(feed, entries: [old_entry, new_entry])
+
+    expect(described_class.call(raw_feed, feed.id)).not_to include(old_entry)
```

Comment:
I think the existing test had a falsely passing test, since it was checking for the array `stories_older_than_3_days` rather than individual items. This highlights a reason I generally avoid things like checking negative inclusion, since the number of items *not* in the list is infinite. It's easy to get a test that is accidentally passing without really testing anything. So, I will instead test what *is* in the list, and often the *whole* list for the sake of precision:

```suggestion
    expect(described_class.call(raw_feed, feed.id)).to eq([new_entry])
```